pkg/api/networkservice/mechanisms/wireguard/helpers.go

```diff
@@ -33,10 +33,14 @@ type Mechanism interface {
 	SrcPublicKey() string
 	// DstPublicKey - destination public key
 	DstPublicKey() string
-	// SrcPort - Source interface listening port
-	SrcPort() int
-	// SrcPort - Destination interface listening port
-	DstPort() int
+	// SrcPrivateKey - source private key
+	SrcPrivateKey() string
+	// DstPrivateKey - destination private key
+	DstPrivateKey() string
+	// SrcPort - source interface listening port
+	SrcPort() uint32
+	// DstPort - destination interface listening port
+	DstPort() uint32
 }
 
 type mechanism struct {
```

Comment:
Let's remove the interface Mechanism  and instead of it make struct public
```suggestion
type Mechanism struct {
```
Example https://github.com/networkservicemesh/api/blob/master/pkg/api/networkservice/mechanisms/kernel/helpers.go#L28